sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientException.java

```diff
@@ -259,7 +261,7 @@ public String toString() {
         return getClass().getSimpleName() + "{" + "error=" + cosmosError + ", resourceAddress='"
                    + resourceAddress + '\'' + ", statusCode=" + statusCode + ", message=" + getMessage()
                    + ", causeInfo=" + causeInfo() + ", responseHeaders=" + responseHeaders + ", requestHeaders="
-                   + requestHeaders + '}';
+                   + filterSensitiveData(requestHeaders) + '}';
```

Comment:
We're still leaking the auth token; just because we override the toString(), it will not prevent someone trapping and dumping the exception object via an object mapper or some other serialization method which ignores the toString() override.

The initial fix that creates a copy of the request header map is a better one if we cannot reuse the original header while removing the sensitive header parts. I checked the code and we only invoke the respective setRequestHeaders() method in exceptional cases only; the only concern here is the memory hit which could go as high as 16k per each request header.